packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,569 @@
+  f_vsn: "1.0.0",
+  status: "DECLINED",
+  reason: "Declined by user."
+}
+
+// Pending - Simple
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // will be included in the requests body
+    params: {}, // will be included in the requests url
+  }
+}
+
+// Pending - First Time with Local
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // included in body of request
+    params: {}, // included as query params on endpoint
+  },
+  local: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "view/frame",
+    endpoint: "https://____", // the iframe that will be rendered,
+    data: {}, // sent to frame when ready
+    params: {}, // included as query params on endpoint
+  }
+}
+```
+
+#### data and params
+
+`data` and `params` are information that the wallet can provide in the service config that FCL will pass back to the service.
+- `params` will be added onto the `endpoint` as query params.
+- `data` will be included in the body of the `HTTP/POST` request or in the `FCL:VIEW:READY:RESPONSE` for a `IFRAME/RPC`, `POP/RPC` or `TAB/RPC`.
+
+# Authentication Service
+
+In the following examples, we'll walk you through the process of building an authentication service.
+
+In FCL, wallets are configured by passing in a wallet provider's authentication URL as the `discovery.wallet` config variable.
+
+As someone who is making an FCL compatible wallet, you will need to make and expose a webpage or API hosted at an authentication endpoint that FCL will use.
+
+```javascript
+// IN APPLICATION
+// configuring fcl to point at a wallet looks like this
+import {config} from "@onflow/fcl"
+
+config({
+  "discovery.wallet": "your-url-that-fcl-will-use-for-authentication",
+  "discovery.wallet.method": "IFRAME/RPC" // Available methods are "IFRAME/RPC", "POP/RPC", "TAB/RPC" or "HTTP/POST"
+})
+```
+
+If the method specified is `IFRAME/RPC`, `POP/RPC` or `TAB/RPC`, then the URL specified as `discovery.wallet` will be rendered as a webpage. Otherwise, if the method specified is `HTTP/POST`, then the authentication process will happen over HTTP requests. (While authentication can be accomplished using any of those service methods, this example will use the `IFRAME/RPC` service method.)
+
+Once the Authentication webpage is rendered, or the API is ready, you then need to tell FCL that it is ready. You will do this by sending a message to FCL, and FCL will send back a message with some additional information that you can use about the application requesting authentication on behalf of the user.
```

Comment:
Can you specify that the following example is using the `IFRAME/RPC` method. And mention that the code that is rendered in the iframe probably has something like following in it.

Reply:
I believe we call this point out here in the sentence in parentheses https://github.com/onflow/flow-js-sdk/pull/753/files/c823457dc251374381103f540e0bf88ebd1ee841#diff-c8ca65624ac7f056410b9805e266ce870c205b739f914da0a8a1f21ea9de817aR198 . We can add another line if you'd like though!

Reply:
Yeah kind of want to really drive the point home.

Reply:
Ok I added in another sentence https://github.com/onflow/flow-js-sdk/pull/753/files#diff-c8ca65624ac7f056410b9805e266ce870c205b739f914da0a8a1f21ea9de817aR202